codalab/common.py

```diff
@@ -279,18 +280,27 @@ def parse_linked_bundle_url(url):
         if url.startswith(StorageURLScheme.AZURE_BLOB_STORAGE.value):
             storage_type = StorageType.AZURE_BLOB_STORAGE.value
             url = url[len(StorageURLScheme.AZURE_BLOB_STORAGE.value) :]
-            storage_account, container, bundle_uuid, contents_file, *remainder = url.split("/", 4)
-            bundle_path = f"{StorageURLScheme.AZURE_BLOB_STORAGE.value}{storage_account}/{container}/{bundle_uuid}/{contents_file}"
+            storage_account, container, *remainder = url.split("/", 2)
+            if len(remainder) and len(remainder[0].split("/")) >= 2:
+                bundle_uuid, contents_file, *remainder = remainder[0].split("/", 2)
+                bundle_path = f"{StorageURLScheme.AZURE_BLOB_STORAGE.value}{storage_account}/{container}/{bundle_uuid}/{contents_file}"
+            else:  # url refers to continer, eg: "azfs://{storage_account}/{container}"
+                bundle_uuid, contents_file, remainder = None, None, []
+                bundle_path = url
         if url.startswith(StorageURLScheme.GCS_STORAGE.value):
             storage_type = StorageType.GCS_STORAGE.value
             url = url[len(StorageURLScheme.GCS_STORAGE.value) :]
-            bucket_name, bundle_uuid, contents_file, *remainder = url.split("/", 3)
-            bundle_path = (
-                f"{StorageURLScheme.GCS_STORAGE.value}{bucket_name}/{bundle_uuid}/{contents_file}"
-            )
-
-        is_archive = contents_file.endswith(".gz") or contents_file.endswith(".tar.gz")
-        is_archive_dir = contents_file.endswith(".tar.gz")
+            bucket_name, *remainder = url.split("/", 1)
+            if len(remainder) and len(remainder[0].split("/")) >= 2:
```

Comment:
this feels pretty complex / difficult to read through.

Maybe it would be simpler if we just did:

```
try:
    bucket_name, bundle_uuid, contents_file, *remainder = url.split("/", 3)
except ValueError:
    # url refers to bucket, e.g. gs://{bucket_name}
    bucket_name, *remainder = url.split("/", 1)
    bundle_uuid, contents_file, remainder = None, None, []
```

What do you think?

Reply:
This would also fit more with the EAFP style recommended for python 🙂 https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/

Reply:
Oh it's much more simpler if we catch the error then deal with it.  I will change it.